services/QuillLMS/spec/controllers/teachers/progress_reports/diagnostic_reports_controller_spec.rb

```diff
@@ -470,4 +470,49 @@
       expect(classroom_unit.reload.assigned_student_ids).to contain_exactly(student1.id, student2.id)
     end
   end
+
+  describe '#assign_diagnostics' do
+    let!(:activity_pack1) { create(:unit_template) }
+    let!(:activity_pack2) { create(:unit_template) }
+    let!(:student1) { create(:student, name: 'Alphabetical A') }
+    let!(:students_classroom1) { create(:students_classrooms, classroom:, student: student1) }
+    let!(:student2) { create(:student, name: 'Alphabetical B') }
+    let!(:students_classroom2) { create(:students_classrooms, classroom:, student: student2) }
+
+    it 'creates a new pack and assigns it if there is no existing pack' do
+      expect do
+        get :assign_diagnostics, params: { classroom_id: classroom.id, diagnostics: [{ unit_template_id: activity_pack1.id, student_ids: [student1.id] }, { unit_template_id: activity_pack2.id, student_ids: [student1.id] }] }
+      end.to change(ClassroomUnit, :count).by(2)
+
+      expect(response).to be_successful
+
+      unit1 = Unit.find_by(user_id: teacher.id, unit_template_id: activity_pack1.id)
+      classroom_unit1 = ClassroomUnit.find_by(unit_id: unit1.id, classroom_id: classroom.id)
+      unit2 = Unit.find_by(user_id: teacher.id, unit_template_id: activity_pack2.id)
+      classroom_unit2 = ClassroomUnit.find_by(unit_id: unit2.id, classroom_id: classroom.id)
+
+      expect(unit1).to be
+      expect(classroom_unit1).to be
+      expect(classroom_unit1.assigned_student_ids).to eq([student1.id])
```

Comment:
It feels like it's worth using `contain_exactly` here instead of `eq` to match the pattern we use below.